execution/executor-benchmark/src/transaction_generator.rs

```diff
@@ -441,6 +451,118 @@ impl TransactionGenerator {
         }
     }
 
+    fn get_connected_grps_transfer_indices(
+        rng: &mut StdRng,
+        num_signer_accounts: usize,
+        block_size: usize,
+        connected_tx_grps: usize,
+    ) -> Vec<(usize, usize)> {
+        let num_accounts_per_grp = num_signer_accounts / connected_tx_grps;
+        // TODO: handle when block_size isn't divisible by connected_tx_grps; an easy
+        //       way to do this is to just generate a few more transactions in the last group
+        let num_txns_per_grp = block_size / connected_tx_grps;
+
+        if num_txns_per_grp >= num_accounts_per_grp {
+            panic!("For the desired workload we want num_accounts_per_grp ({}) > num_txns_per_grp ({})", num_accounts_per_grp, num_txns_per_grp);
+        }
+
+        (0..connected_tx_grps)
+            .flat_map(|grp_idx| {
+                let accounts_st_idx = grp_idx * num_accounts_per_grp;
+                let accounts_end_idx = accounts_st_idx + num_accounts_per_grp - 1;
+                let mut unused_indices: Vec<_> = (accounts_st_idx..=accounts_end_idx).collect();
+                unused_indices.shuffle(rng);
+                let mut used_indices: Vec<_> =
+                    vec![unused_indices.pop().unwrap(), unused_indices.pop().unwrap()];
+                let mut transfer_indices: Vec<(_, _)> = vec![(used_indices[0], used_indices[1])];
+
+                for _ in 1..num_txns_per_grp {
+                    // index1 is always from used_indices, so that all the txns are connected
+                    let mut index1 = used_indices[rng.gen_range(0, used_indices.len())];
+
+                    // index2 is either from used_indices or unused_indices
+                    let mut index2;
+                    let rnd = rng.gen_range(0, used_indices.len() + unused_indices.len());
```

Comment:
I am not sure if this will result in a desired random distribution. As the used indices increases, the probability of choosing the used index will grow larger. May be we should just flip a coin and decide if to pick from used vs unused?

Reply:
I see your point. By flipping a coin we decide whether we have one side or both sides connected with equal probabilities. Will add the logic